realm/realm-library/src/main/java/io/realm/RealmObjectSchema.java

```diff
@@ -462,6 +462,22 @@ public void setEmbedded(boolean embedded) {
         }
     }
 
+    /**
+     * Retrieves the type of a given property.
+     *
+     * @param property the field for which we want to know the type.
+     * @return the name of the property.
+     * @throws IllegalStateException if the given property is not found in the schema.
+     */
+    public String getPropertyType(String property) {
```

Comment:
This implementation is a bit problematic to have in `RealmObjectSchema` .. methods in here should work for both `DynamicRealm` and `Realm` instances, but `ColumnInfo` doesn't exist for DynamicRealms. I think this implementation should be in `ImmutableRealmObjectSchema` instead and we 
need another implementation for the `MutableRealmObjectSchema`.

Since this is part of the public API we also need tests specifically for this outside the context of `createEmbeddedObject`